api.py

```diff
@@ -0,0 +1,246 @@
+CLIENT = None
+
+@api.errorhandler(Exception)
+def handle_unexpected_error(error):
+    status_code = 500
+    response = {
+        'error': str(error)
+    }
+    logger.exception(error)
+    return jsonify(response), status_code
+
+@api.errorhandler(JsonValidationError)
+def handle_validation_error(e):
+    return jsonify({
+        'error': e.message,
+        'errors': [validation_error.message for validation_error  in e.errors],
+    })
+
+@api.route('/devices', methods=['GET'])
+def list_devices():
+    return jsonify(commands.enumerate())
+
+@api.route('/prompt', methods=['POST'])
+@schema.validate({
+    'required': ['path'],
+    'properties': {
+        'path':{ 'type': 'string' },
+    },
+})
+def prompt_device():
+    global CLIENT
+    path = request.json['path']
+    client, device = get_client_and_device(path)
+    if device.get('needs_pin_sent'):
+        CLIENT = client
+        client.prompt_pin()
+    elif device.get('needs_password_sent'):
+        # TODO this is for bitbox. not sure how to implement ...
+        raise NotImplementedError()
+    else:
+        raise Exception(f'Device with path {device["path"]} already unlocked')
+    return jsonify({})  # FIXME: what to do here when there's nothing to return
+
+@api.route('/unlock', methods=['POST'])
+@schema.validate({
+    'required': ['path'],
+    'properties': {
+        'pin': { 'type': 'string' },  # trezor
+        'password': { 'type': 'string' },  # bitbox
+        'path':{ 'type': 'string' },
+    },
+})
+def unlock_device():
+    '''Prompt every device that's plugged in'''
+    # more validation
+    pin = request.json.get('pin')
+    password = request.json.get('password')
+    assert pin or password   # FIXME do in json schema
+
+    # unlock device and return response
+    global CLIENT
+    result = CLIENT.send_pin(pin)
+    if result['success']:
+        del CLIENT
+        return jsonify({})  # FIXME
+    else:
+        raise Exception('Failed to unlock device')
+
+
+@api.route('/wallets', methods=['GET'])
+def list_wallets():
+    # TODO: does this include addresses?
+    wallets = get_wallets()
+    # FIXME: probably shouldn't include xpubs in this response?
+    wallet_dicts = [wallet.to_dict(True) for wallet in wallets]
+    return jsonify(wallet_dicts)
+
+@api.route('/wallets', methods=['POST'])
+@schema.validate({
+    'required': ['name', 'm', 'n'],
+    'properties': {
+        'name': { 'type': 'string' },
+        'm': { 'type': 'integer' },
+        'n': { 'type': 'integer' },
+    },
+})
+def create_wallet():
+    wallet = MultisigWallet.create(**request.json)
+    return jsonify({})  # FIXME
+
+@api.route('/signers', methods=['POST'])
+@schema.validate({
+    'required': ['wallet_name', 'signer_name', 'fingerprint'],
+    'properties': {
+        'wallet_name': { 'type': 'string' },
+        'signer_name': { 'type': 'string' },
+        # FIXME: or path???
+        'fingerprint': { 'type': 'string' },  # FIXME: regex
```

Comment:
Looks like fingerprint and path are used inconsistently. Maybe we should just have one arg (id?) and allow it to be either, since it looks like `get_client_and_device` doesn't much care which it gets.

Reply:
I like that idea